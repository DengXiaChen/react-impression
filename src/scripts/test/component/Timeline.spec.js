import React from 'react';
import { mount, shallow } from 'enzyme';
import { Timeline, Icon } from '../../components/impression';

const setup = () => {
    const temp = (
        <Timeline>
            <Timeline.Item
                dot={<Icon type="fa-play-circle-o" />}>
                1999 疯狂世界
            </Timeline.Item>
            <Timeline.Item unreachable>2000 爱情万岁</Timeline.Item>
            <Timeline.Item>2001 候鸟</Timeline.Item>
            <Timeline.Item>2002 摇滚本事</Timeline.Item>
        </Timeline>
    );
    const wrapper = mount(temp);

    const wrapper1 = shallow(temp);

    return {
        wrapper,
        wrapper1,
    };
};

describe('Timeline', () => {
    const { wrapper, wrapper1 } = setup();

    it('render', () => {
        expect(wrapper.find('.timeline').exists()).toBe(true);
    });

    it('item', () => {
        expect(wrapper.find('.timeline-item')).toHaveLength(4);
    });

    it('dot', () => {
        expect(wrapper.find('Icon').exists()).toBe(true);
    });

    it('unreachable', () => {
        expect(wrapper.find('.timeline-item-unreachable').text()).toBe('2000 爱情万岁');
    });

    it('ui', () => {
        expect(wrapper1).toMatchSnapshot();
    });
});
