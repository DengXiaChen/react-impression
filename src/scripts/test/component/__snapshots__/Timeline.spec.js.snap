// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="timeline"
>
    <TimelineItem
        dot={
            <Icon
                left={false}
                right={false}
                type="fa-play-circle-o"
            />
        }
    >
        1999 疯狂世界
    </TimelineItem>
    <TimelineItem
        unreachable={true}
    >
        2000 爱情万岁
    </TimelineItem>
    <TimelineItem>
        2001 候鸟
    </TimelineItem>
    <TimelineItem>
        2002 摇滚本事
    </TimelineItem>
</ul>,
  "nodes": Array [
    <ul
      className="timeline"
>
      <TimelineItem
            dot={
                  <Icon
                        left={false}
                        right={false}
                        type="fa-play-circle-o"
                  />
            }
      >
            1999 疯狂世界
      </TimelineItem>
      <TimelineItem
            unreachable={true}
      >
            2000 爱情万岁
      </TimelineItem>
      <TimelineItem>
            2001 候鸟
      </TimelineItem>
      <TimelineItem>
            2002 摇滚本事
      </TimelineItem>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Timeline>
        <TimelineItem
                dot={
                        <Icon
                                left={false}
                                right={false}
                                type="fa-play-circle-o"
                        />
                }
        >
                1999 疯狂世界
        </TimelineItem>
        <TimelineItem
                unreachable={true}
        >
                2000 爱情万岁
        </TimelineItem>
        <TimelineItem>
                2001 候鸟
        </TimelineItem>
        <TimelineItem>
                2002 摇滚本事
        </TimelineItem>
</Timeline>,
      "_debugID": 29,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <TimelineItem
              dot={
                            <Icon
                                          left={false}
                                          right={false}
                                          type="fa-play-circle-o"
                            />
              }
>
              1999 疯狂世界
</TimelineItem>,
            <TimelineItem
              unreachable={true}
>
              2000 爱情万岁
</TimelineItem>,
            <TimelineItem>
              2001 候鸟
</TimelineItem>,
            <TimelineItem>
              2002 摇滚本事
</TimelineItem>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="timeline"
>
          <TimelineItem
                    dot={
                              <Icon
                                        left={false}
                                        right={false}
                                        type="fa-play-circle-o"
                              />
                    }
          >
                    1999 疯狂世界
          </TimelineItem>
          <TimelineItem
                    unreachable={true}
          >
                    2000 爱情万岁
          </TimelineItem>
          <TimelineItem>
                    2001 候鸟
          </TimelineItem>
          <TimelineItem>
                    2002 摇滚本事
          </TimelineItem>
</ul>,
        "_debugID": 30,
        "_renderedOutput": <ul
          className="timeline"
>
          <TimelineItem
                    dot={
                              <Icon
                                        left={false}
                                        right={false}
                                        type="fa-play-circle-o"
                              />
                    }
          >
                    1999 疯狂世界
          </TimelineItem>
          <TimelineItem
                    unreachable={true}
          >
                    2000 爱情万岁
          </TimelineItem>
          <TimelineItem>
                    2001 候鸟
          </TimelineItem>
          <TimelineItem>
                    2002 摇滚本事
          </TimelineItem>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Timeline>
    <TimelineItem
        dot={
            <Icon
                left={false}
                right={false}
                type="fa-play-circle-o"
            />
        }
    >
        1999 疯狂世界
    </TimelineItem>
    <TimelineItem
        unreachable={true}
    >
        2000 爱情万岁
    </TimelineItem>
    <TimelineItem>
        2001 候鸟
    </TimelineItem>
    <TimelineItem>
        2002 摇滚本事
    </TimelineItem>
</Timeline>,
}
`;
